src/main/java/org/candlepin/subscriptions/tally/admin/InternalTallyResource.java

```diff
@@ -20,26 +20,73 @@
  */
 package org.candlepin.subscriptions.tally.admin;
 
+import java.time.OffsetDateTime;
+import javax.ws.rs.BadRequestException;
+import lombok.extern.slf4j.Slf4j;
+import org.candlepin.subscriptions.ApplicationProperties;
+import org.candlepin.subscriptions.resource.ResourceUtils;
 import org.candlepin.subscriptions.tally.MarketplaceResendTallyController;
+import org.candlepin.subscriptions.tally.TallySnapshotController;
 import org.candlepin.subscriptions.tally.admin.api.InternalApi;
 import org.candlepin.subscriptions.tally.admin.api.model.TallyResend;
 import org.candlepin.subscriptions.tally.admin.api.model.TallyResendData;
 import org.candlepin.subscriptions.tally.admin.api.model.UuidList;
 import org.candlepin.subscriptions.tally.billing.RemittanceController;
+import org.candlepin.subscriptions.tally.job.CaptureSnapshotsTaskManager;
+import org.candlepin.subscriptions.util.ApplicationClock;
+import org.candlepin.subscriptions.util.DateRange;
 import org.springframework.stereotype.Component;
 
 /** This resource is for exposing administrator REST endpoints for Tally. */
 @Component
+@Slf4j
 public class InternalTallyResource implements InternalApi {
 
+  private final ApplicationClock clock;
+  private final ApplicationProperties applicationProperties;
   private final MarketplaceResendTallyController resendTallyController;
   private final RemittanceController remittanceController;
+  private final TallySnapshotController tallySnapshotController;
+  private final CaptureSnapshotsTaskManager snapshotsTaskManager;
 
   public InternalTallyResource(
+      ApplicationClock clock,
+      ApplicationProperties applicationProperties,
       MarketplaceResendTallyController resendTallyController,
-      RemittanceController remittanceController) {
+      RemittanceController remittanceController,
+      TallySnapshotController tallySnapshotController,
+      CaptureSnapshotsTaskManager snapshotsTaskManager) {
+    this.clock = clock;
+    this.applicationProperties = applicationProperties;
     this.resendTallyController = resendTallyController;
     this.remittanceController = remittanceController;
+    this.tallySnapshotController = tallySnapshotController;
+    this.snapshotsTaskManager = snapshotsTaskManager;
+  }
+
+  @Override
+  public void performHourlyTallyForAccount(
+      String account,
+      OffsetDateTime start,
+      OffsetDateTime end,
+      Boolean xRhSwatchSynchronousRequest) {
+    DateRange range = new DateRange(start, end);
+    if (!clock.isHourlyRange(range)) {
```

Comment:
I'd really like to do this with a validation annotation.  See `ParameterDurationValidator` for an example of how to apply a validation to two parameters.  Ideally, we could also apply the validation annotation to the JMX bean as well.  I'll be happy to help on this too.  I've done a fair bit with the validation API.

Reply:
I gave this a shot and got down in the weeds so I pulled the plug on it. If you feel strongly about this, I'd say you add a new card to the backlog and do it another time since this PR seems to be dragging on. Thoughts?